policy.go

```diff
@@ -0,0 +1,74 @@
+package proxyproto
+
+import "net"
+
+// PolicyFunc can be used to decide whether to trust the PROXY info from
+// upstream. If set, the connecting address is passed in as an argument.
+//
+// See below for the different policies.
+//
+// In case an error is returned the connection is denied.
+type PolicyFunc func(upstream net.Addr) (Policy, error)
+
+// Policy defines how a connection with a PROXY header address is treated.
+type Policy int
+
+const (
+	// USE address from PROXY header
+	USE Policy = iota
+	// IGNORE address from PROXY header, but accept connection
+	IGNORE
+	// REJECT connection when PROXY header is sent
+	// Note: even though the first read on the connection returns an error if
+	// a PROXY header is present, subsequent reads do not. It is the task of
+	// the code using the connection to handle that case properly.
+	REJECT
+	// REQUIRE connection to send PROXY header, reject if not present
+	// Note: even though the first read on the connection returns an error if
+	// a PROXY header is not present, subsequent reads do not. It is the task
+	// of the code using the connection to handle that case properly.
+	REQUIRE
+)
+
+// LaxWhiteListPolicy returns a PolicyFunc which decides whether the
+// upstream ip is allowed to send a proxy header based on a list of allowed
+// IP addresses. In case upstream IP is not in list the proxy header will
+// be ignored.
+func LaxWhiteListPolicy(allowed []string) PolicyFunc {
```

Comment:
Here and in the policy below my suggestion would be to allow CIDRs rather than IP addresses (where the CIDRs can be e.g. /32). It can make some workflows a bit easier because you can know that your LBs will all be in the same subnet but if they can come and go (e.g. if they're being switched out for upgrades) if new ones come up they will be automatically trusted.

It's not a big deal, but it is a nicety. Stdlib ParseCIDR unfortunately _requires_ a CIDR; an alternative could be https://godoc.org/github.com/hashicorp/go-sockaddr#NewIPAddr. There are also functions in there that can check whether a given address is contained in the parsed subnet.

Reply:
I don't like adding a dependency just for that use case, especially as this library doesn't have any other yet. I'd rather try to check if the string contains a slash, and then treat it as a CIDR using `net.ParseCIDR()`, and as IP address otherwise.

Reply:
Sounds fine, I did forget about IPNet.Contains in stdlib.

Reply:
I like the idea of supporting CIDRs, as well. Thank you, @mikey179!

Reply:
Added support for CIDRs in 3eee238. The policy creating functions now return an error in case an invalid CIDR or IP address is provided. In case you don't want to explicitly check the errors new Must* functions were added, so if your program reads this from a configuration you can be sure it panics if the configuration is wrong.